core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -405,6 +427,23 @@ private[spark] class MemoryStore(
 
   def clear(): Unit = memoryManager.synchronized {
     entries.synchronized {
+      val entryManualCloseTasks = ArrayBuffer.empty[Future[Unit]]
+      entries.values.asScala.foreach {
+        case e: DeserializedMemoryEntry[_] => e.value.foreach {
+          case o: AutoCloseable =>
+            entryManualCloseTasks += Future {
```

Comment:
Not bad; I am a little concerned about the extra complexity and which thread pool this gets called in, maybe creating its own bottleneck. But you could argue it's better than waiting in clear(). Are any of the objects closed here expensive to close? I didn't check.

Reply:
@srowen , yes, your concern totally makes sense, now I changed to only use one Future to cleanup all, so cleanup won't start too much threads, and I added a 10ms wait in each close [UnitTest Link](https://github.com/apache/spark/pull/32534/files#diff-c4f6f7a030459e4a28c597761512bf4f02d4a213818da66525e0ba9c0327c47aR628-R637) to pretend as an expensive close.